controllers/configurationpolicy_controller.go

```diff
@@ -2798,6 +2802,36 @@ func handleKeys(
 		}
 	}
 
+	// If the complianceType is "mustonlyhave", then compare the existing object's keys,
+	// skipping over: previously compared keys, metadata, and status.
+	if compType == "mustonlyhave" {
+		for key := range existingObj.Object {
+			if handledKeys[key] || key == "status" || key == "metadata" {
+				continue
+			}
+
+			// check key for mismatch
+			errorMsg, keyUpdateNeeded, mergedObj, skipped := handleSingleKey(
+				key, desiredObj, existingObjectCopy, compType, zeroValueEqualsNil,
+			)
+			if errorMsg != "" {
+				log.Info(errorMsg)
+
+				return true, errorMsg, true, statusMismatch
+			}
+
+			// If the mergedObj is nil, we'll use the value to erase it from the existing object.
+			if skipped {
+				continue
+			}
+
+			if keyUpdateNeeded {
+				existingObj.Object[key] = mergedObj
+				updateNeeded = true
+			}
```

Comment:
Can this be simplified to the following? I'm think that if the keys weren't present on `desiredObj.Object` but exist on `existingObj.Object`, then I think you always just want to delete the key.
```suggestion
			delete(existingObj.Object, key)
			updateNeeded = true
```

Reply:
Ah, true. That's way cleaner--thanks!

Reply:
I guess I was thinking of inheriting some of the logic in `handleSingleKey`, but deleting it seems right. The dry-run ought to hopefully take care of any edge cases.